src/tools/run-make-support/src/external_deps/c_build.rs

```diff
@@ -25,3 +26,33 @@ pub fn build_native_static_lib(lib_name: &str) -> PathBuf {
     llvm_ar().obj_to_ar().output_input(&lib_path, &obj_file).run();
     path(lib_path)
 }
+
+/// Builds a dynamic lib (`.dll` on Windows, `.dylib` on OSX and `.so` for Linux) with the given
+/// name.
+#[track_caller]
+pub fn build_native_dynamic_lib(lib_name: &str) -> PathBuf {
+    let obj_file = if is_msvc() { format!("{lib_name}") } else { format!("{lib_name}.o") };
+    let src = format!("{lib_name}.c");
+    let lib_path = dynamic_lib_name(lib_name);
+    if is_msvc() {
+        cc().arg("-c").out_exe(&obj_file).input(src).run();
+    } else {
+        cc().arg("-v").arg("-c").out_exe(&obj_file).input(src).run();
+    };
+    let obj_file = if is_msvc() { format!("{lib_name}.obj") } else { format!("{lib_name}.o") };
```

Comment:
Obviously not a job for here, but maybe run-make should switch from a `&str`-based API to `Path`-based so these turn into e.g. `path_buf.set_extension(...)` and `path.extension().unwrap()`, rather than manipulating the strings directly. And then just take `impl AsRef<OsStr>` everywhere so `&str` still works.

Reply:
Yeah, feel free to leave a FIXME here for me.